anomalib/data/ucsd_ped.py

```diff
@@ -0,0 +1,269 @@
+
+    The files are expected to follow the structure:
+        path/to/dataset/category/split/video_id/image_filename.tif
+        path/to/dataset/category/split/video_id_gt/mask_filename.bmp
+
+    Args:
+        root (Path): Path to dataset
+        split (Optional[Union[Split, str]], optional): Dataset split (ie., either train or test). Defaults to None.
+
+    Example:
+        The following example shows how to get testing samples from UCSDped2 category:
+
+        >>> root = Path('./UCSDped')
+        >>> category = 'UCSDped2'
+        >>> path = root / category
+        >>> path
+        PosixPath('UCSDped/UCSDped2')
+
+        >>> samples = make_ucsd_dataset(path, split='test')
+        >>> samples.head()
+           root             folder image_path                    mask_path                         split
+        0  UCSDped/UCSDped2 Test   UCSDped/UCSDped2/Test/Test001 UCSDped/UCSDped2/Test/Test001_gt  test
+        1  UCSDped/UCSDped2 Test   UCSDped/UCSDped2/Test/Test002 UCSDped/UCSDped2/Test/Test002_gt  test
+        ...
+
+    Returns:
+        DataFrame: an output dataframe containing samples for the requested split (ie., train or test)
+    """
+    folders = [filename for filename in sorted(Path(path).glob("*/*")) if os.path.isdir(filename)]
+    folders = [folder for folder in folders if len(list(folder.glob("*.tif"))) > 0]
+
+    samples_list = [(str(path),) + folder.parts[-2:] for folder in folders]
+    samples = DataFrame(samples_list, columns=["root", "folder", "image_path"])
+
+    samples.loc[samples.folder == "Test", "mask_path"] = samples.image_path.str.split(".").str[0] + "_gt"
+    samples.loc[samples.folder == "Test", "mask_path"] = samples.root + "/" + samples.folder + "/" + samples.mask_path
+    samples.loc[samples.folder == "Train", "mask_path"] = ""
+
+    samples["image_path"] = samples.root + "/" + samples.folder + "/" + samples.image_path
+
+    samples.loc[samples.folder == "Train", "split"] = "train"
+    samples.loc[samples.folder == "Test", "split"] = "test"
+
+    if split:
+        samples = samples[samples.split == split]
+        samples = samples.reset_index(drop=True)
+
+    return samples
+
+
+class UCSDpedClipsIndexer(ClipsIndexer):
+    """Clips class for UCSDped dataset."""
+
+    def get_mask(self, idx) -> Optional[Tensor]:
+        """Retrieve the masks from the file system."""
+
+        video_idx, frames_idx = self.get_clip_location(idx)
+        mask_folder = self.mask_paths[video_idx]
+        if mask_folder == "":  # no gt masks available for this clip
+            return None
+        frames = self.clips[video_idx][frames_idx]
+
+        mask_frames = sorted(glob.glob(mask_folder + "/*"))
+        mask_paths = [mask_frames[idx] for idx in frames.int()]
+
+        masks = torch.stack([Tensor(cv2.imread(mask_path, flags=0)) / 255.0 for mask_path in mask_paths])
+        return masks
+
+    def _compute_frame_pts(self) -> None:
```

Comment:
What does `pts` mean?

Reply:
This method signature is inherited from torchvision's `VideoClips` class. Documentation is a bit poor, but it seems to stand for "presentation times" or "presentation time stamps", which just means the collection of time stamps in number of frames for each video/video_clip.